spec/lib/solidus_shipstation/api/batch_syncer_spec.rb

```diff
@@ -0,0 +1,202 @@
+RSpec.describe SolidusShipstation::Api::BatchSyncer do
+  include ActiveSupport::Testing::TimeHelpers
+
+  describe '.from_config' do
+    it 'creates a syncer with the configured API client' do
+      client = instance_double(SolidusShipstation::Api::Client)
+      allow(SolidusShipstation::Api::Client).to receive(:from_config).and_return(client)
+
+      batch_syncer = described_class.from_config
+
+      expect(batch_syncer).to have_attributes(client: client)
+    end
+  end
+
+  describe '#call' do
+    context 'when the API call is successful' do
+      context 'when the sync operation succeeded' do
+        it 'updates the ShipStation data on the shipment' do
+          freeze_time do
+            shipment = instance_spy('Spree::Shipment', number: 'H123456')
+            api_client = instance_double(SolidusShipstation::Api::Client).tap do |client|
+              allow(client).to receive(:bulk_create_orders).with([shipment]).and_return({
```

Comment:
Maybe this could be written as:

```
              allow(client).to receive(:bulk_create_orders).with([shipment]).and_return({
                'results' => [{
                  'orderNumber' => shipment.number,
                  'success' => true,
                  'orderId' => '123456',
                }]
              })
```